text/0000-colocated-tests.md

```diff
@@ -0,0 +1,191 @@
+
+### Addons, in-repo Addons and Engines
+
+For normal addons, living as their own entire project, the lookup rules for apps will apply.
+For in-repo addons and engines, there is the possibility that the implementation of this feature would enable things not possible today.
+For example, if the "test finder" followed the glob: `**/*(\.test.js|-test.js)`, (excluding node\_modules, etc), that would include tests that are in in-repo addons and engines,
+which today, do not support their own tests -- the in-repo addon or engine still wouldn't support their own tests, exactly, as they would run as part of the parent
+app's test suite. This might be desired behavior as it would allow the in-repo addon and engine to finally co-locate their tests with the behavior that those units implement.
+
+
+### Reference: Where do the existing tests get migrated to?
+While tests in `{app,addon}/` could live anywhere, these are to be the default locations for the generators to use.
+
+#### Route
+
+previously:
+ - `{app,addon}/routes/{route-name}.js`
+ - `tests/unit/{route-name}-test.js`
+now:
+ - `{app,addon}/routes/{route-name}.js`
+ - `{app,addon}/routes/{route-name}.test.js`
+
+#### Service
+
+previously:
+ - `{app,addon}/services/{service-name}.js`
+ - `tests/unit/{service-name}-test.js`
+now:
+ - `{app,addon}/service/{service-name}.js`
+ - `{app,addon}/service/{service-name}.test.js`
+
+#### Controller
+
+previously:
+ - `{app,addon}/controllers/{controller-name}.js`
+ - `tests/unit/{controller-name}-test.js`
+now:
+ - `{app,addon}/controllers/{controller-name}.js`
+ - `{app,addon}/controllers/{controller-name}.test.js`
+
+#### Component
+
+previously:
+ - `{app,addon}/components/{component-name}.js`
+ - `{app,addon}/components/{component-name}.hbs`
+ - `tests/integration/components/{component-name}-test.js`
+now:
+ - `app/components/{component-name}.js`
+ - `app/components/{component-name}.hbs`
+ - `app/components/{component-name}.test.js`
+
+or using `"componentStructure": "nested"`:
+ - `{app,addon}/components/{component-name}/index.js`
+ - `{app,addon}/components/{component-name}/index.hbs`
+ - `{app,addon}/components/{component-name}/index.test.js`
+
+#### Helpers
+
+previously:
+ - `{app,addon}/helpers/{helper-name}.js`
+ - `tests/integration/helpers/{helper-name}-test.js`
+now:
+ - `{app,addon}/helpers/{helper-name}.js`
+ - `{app,addon}/helpers/{helper-name}.test.js`
+
+#### Utils
+
+previously:
+ - `{app,addon}/utils/{util-name}.js`
+ - `tests/unit/{util-name}-test.js`
+now:
+ - `{app,addon}/utils/{util-name}.js`
+ - `{app,addon}/utils/{util-name}.test.js`
+
+#### Acceptance Tests
+
+Acceptance tests do not have a core affiliation with any other file, so co-locating them _may_ not make sense, depending on what a particular test file is doing.
+
+While tests may live anywhere in the `{app,addon}` directory, the default location for acceptance tests will still be `tests/acceptance/`.
```

Comment:
Engines do have a weird setup with the dummy app, so that might be the co-location in that case.
Also I'm not a big fan of the dummy app being used for docs (e.g. ec-addon-docs) and is loaded in (part of) tests... can sb free it up from the identity crisis? :D